models/Product.ts

```diff
@@ -0,0 +1,26 @@
+import mongoose from 'mongoose'
+import { ProductCardProps } from '~/components/ProductCard'
+
+const ProductSchema = new mongoose.Schema<ProductCardProps>({
```

Comment:
I don't think using the component-specific props object in the schema would be a good approach.
Other than that, this doesn't seem to be the most type-safe approach. Once you call the contractor of this class, the returned object will be of type `any`.

Consider using a ORM like Drizzle (for SQL databases) or Prisma (if you need to use Mongo): https://www.prisma.io/docs/orm/overview/databases/mongodb

Reply:
Will using mongoose schemas be enough for this ?

```typescript
import mongoose from 'mongoose'
import { ProductCardProps } from '~/components/ProductCard'

const ProductSchema = new mongoose.Schema<ProductCardProps>({
	id: { type: String, required: true, unique: true },
	name: { type: String, required: true, unique: true },
	price: { type: Number, required: true },
	available: { type: Number, required: true },
	popularity: { type: Number, required: true, default: 0 },
	rating: { type: Number, required: true, default: 0 },
	discount: { type: Number, required: true },
	categoryName: { type: String, required: true },
	subCategoryName: { type: String, default: null },
	categoryId: { type: Number, required: true },
	subCategoryId: {
		type: Number,
		default: null,
		required: false,
	},
	imgURL: { type: String, required: true },
	description: { type: String, required: true },
})

const Product = mongoose.models.Product || mongoose.model('Product', ProductSchema)

export default Product
```